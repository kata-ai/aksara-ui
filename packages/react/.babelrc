{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": {
          "browsers": ["last 2 chrome versions", "last 2 firefox versions"]
        }
      }
    ],
    "@babel/preset-react",
    "@babel/preset-typescript"
  ],
  "plugins": [
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "babel-plugin-styled-components"
  ],
  "env": {
    "test": {
      "plugins": ["@babel/plugin-transform-modules-commonjs"]
    }
  }
}
