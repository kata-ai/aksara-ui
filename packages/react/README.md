# @aksara-ui/react

> React implementation of Aksara Design System, from Kata.ai.

This package is part of the Aksara monorepo. Please read the README in [the Aksara repository](https://github.com/kata-ai/aksara-ui) for detailed documentation about this package.
