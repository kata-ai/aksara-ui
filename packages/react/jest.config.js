module.exports = {
  collectCoverageFrom: [
    'src/**/*.{ts,tsx}',
    '!src/**/*.stories.{ts,tsx}',
    '!src/**/*.story.{ts,tsx}',
    '!src/**/index.{ts,tsx}',
    '!src/stories/**/*.{ts,tsx}',
    '!src/utils/storybook/**/*.{ts,tsx}',
    '!src/index.{ts,tsx}',
  ],
  coverageReporters: ['json', 'lcov'],
  moduleDirectories: ['node_modules', '<rootDir>/src'],
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json'],
  modulePathIgnorePatterns: ['lib', 'lib-es6'],
  setupFilesAfterEnv: ['<rootDir>/test/setupTests.ts'],
  testPathIgnorePatterns: ['<rootDir>/dist/'],
  transform: {
    '^.+\\.(js|jsx|ts|tsx|mjs)$': 'babel-jest',
    '^.+\\.css$': '<rootDir>/config/jest/cssTransform.js',
    '^(?!.*\\.(js|jsx|ts|tsx|mjs|css|json)$)': '<rootDir>/config/jest/fileTransform.js',
  },
};
