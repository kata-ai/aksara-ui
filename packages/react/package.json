{
  "name": "@aksara-ui/react",
  "version": "1.0.1-canary.4",
  "description": "React implementation of Aksara Design System, from Kata.ai.",
  "main": "dist/index.js",
  "module": "dist/core.esm.js",
  "typings": "dist/index.d.ts",
  "aksara:src": "./src/index.ts",
  "repository": "https://github.com/kata-ai/wicara.git",
  "author": "Kata.ai <dev@kata.ai>",
  "license": "Apache-2.0",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "tsdx build --transpileOnly --tsconfig tsconfig.build.json",
    "build-storybook": "build-storybook --quiet",
    "clean": "rimraf dist",
    "format": "prettier --write \"src/**/*.{js,jsx,ts,tsx}\"",
    "lint": "eslint \"src/**/*.{js,jsx,ts,tsx}\"",
    "prepublish": "yarn build",
    "report-coverage": "codecov -f ./coverage/*.json",
    "start": "tsdx watch --tsconfig tsconfig.build.json --verbose",
    "storybook": "start-storybook -p 9001",
    "storybook:build": "build-storybook --quiet",
    "test": "jest",
    "test:coverage": "jest --coverage",
    "type-check": "tsc --emitDeclarationOnly"
  },
  "dependencies": {
    "@aksara-ui/core": "^1.0.1-canary.1",
    "@popperjs/core": "^2.10.2",
    "@reach/accordion": "^0.16.1",
    "@reach/visually-hidden": "^0.16.0",
    "@styled-system/css": "^5.1.5",
    "@styled-system/should-forward-prop": "^5.1.5",
    "@styled-system/theme-get": "^5.1.2",
    "@types/styled-system": "^5.1.13",
    "@types/styled-system__css": "^5.0.16",
    "@types/styled-system__theme-get": "^5.0.1",
    "clsx": "^1.1.1",
    "csstype": "^3.0.9",
    "deepmerge": "^4.2.2",
    "focus-trap": "^6.6.0",
    "polished": "^4.1.3",
    "react-fast-compare": "^3.2.0",
    "react-popper": "^2.2.5",
    "react-popper-tooltip": "^4.3.0",
    "react-transition-group": "^4.4.2",
    "styled-system": "^5.1.4",
    "tslib": "^2.3.1"
  },
  "peerDependencies": {
    "@aksara-ui/icons": "^1.0.0-canary.0",
    "react": "^16.10.0 || ^17.0.0",
    "react-dom": "^16.10.0 || ^17.0.0",
    "styled-components": "^5.1.0"
  },
  "devDependencies": {
    "@aksara-ui/icons": "^1.0.0",
    "@babel/core": "^7.15.0",
    "@storybook/addon-actions": "^6.3.7",
    "@storybook/addon-essentials": "^6.3.7",
    "@storybook/addon-knobs": "^6.3.0",
    "@storybook/addon-links": "^6.3.7",
    "@storybook/react": "^6.3.7",
    "@storybook/theming": "^6.3.7",
    "@testing-library/jest-dom": "^5.14.1",
    "@testing-library/react": "^12.0.0",
    "@types/jest": "^26.0.24",
    "@types/react-transition-group": "^4.4.2",
    "@types/styled-components": "^5.1.12",
    "babel-loader": "^8.2.2",
    "babel-plugin-styled-components": "^1.13.2",
    "babel-preset-react-app": "^10.0.0",
    "codecov": "^3.8.3",
    "copee": "^1.0.6",
    "jest": "^26.6.3",
    "jest-styled-components": "^7.0.5",
    "react": "^17.0.2",
    "react-docgen-typescript-loader": "^3.6.0",
    "react-dom": "^17.0.2",
    "styled-components": "^5.3.0"
  }
}
