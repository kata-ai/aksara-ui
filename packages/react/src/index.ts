export * from './aksara-provider';
export * from './components';
export * from './layout';
export * from './theme';
export * from './typography';
export * from '@aksara-ui/helpers';
