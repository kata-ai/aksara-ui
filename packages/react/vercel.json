{
  "version": 2,
  "headers": [
    {
      "source": "/static/(.*)",
      "headers": [
        {
          "key": "Cache-Control",
          "value": "public, max-age=0, must-revalidate"
        }
      ]
    },
    {
      "source": "/(.*)",
      "headers": [
        {
          "key": "Cache-Control",
          "value": "s-maxage=0"
        }
      ]
    }
  ]
}
