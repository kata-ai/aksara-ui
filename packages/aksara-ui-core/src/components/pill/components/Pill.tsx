import * as React from 'react';
import styled, { css } from 'styled-components';
import { IconClose } from '@aksara-ui/icons';

import { Box, BoxProps } from '../../../foundations';
import { useComponentStyles } from '../../../system';
import { IconButton } from '../../button';
import { theme } from '../../../theme';

export interface PillProps extends Omit<BoxProps, 'size'> {
  /** The variant options available for a pill. */
  variant?: 'default' | 'neutral' | 'info' | 'warning' | 'critical' | 'success';
  /** Size of the pill. Defaults to 'md' */
  size?: 'sm' | 'md' | 'lg';
  /** Additional CSS classes to give to the component. */
  className?: string;
  /** Additional CSS properties to give to the component. */
  style?: React.CSSProperties;
  /** Triggers when the Close button is clicked. */
  onClose?: () => void;
}

const CloseButton = styled(IconButton)<{ outline?: boolean }>`
  background-color: rgba(49, 63, 78, 0.2);
  color: ${theme.colors.grey03};
  border-radius: 0;
  outline: none;
  height: 100%;

  svg {
    width: 16px;
    height: 16px;
  }

  &:focus,
  &:active {
    border-width: 0 !important;
    box-shadow: none !important;
  }

  ${({ outline }) =>
    outline &&
    css`
      color: ${theme.colors.grey08};
    `}
`;

const Pill: React.FC<PillProps> = ({
  className,
  style,
  children,
  size = 'sm',
  variant = 'default',
  onClose,
  sx,
  ...rest
}) => {
  const boxStyles = useComponentStyles('pill', { variant });
  const buttonStyles = useComponentStyles('pill', { size, buttonVariant: variant });

  return (
    <Box
      display="flex"
      flexDirection="row"
      borderRadius="4px"
      justifyContent="space-between"
      style={style}
      className={className}
      sx={{ ...boxStyles, ...sx }}
      {...rest}
    >
      <Box as="span" role="button" sx={{ ...buttonStyles, ...sx }}>
        {children}
      </Box>
      {onClose && (
        <CloseButton onClick={onClose} outline={variant === 'neutral'}>
          <IconClose fill="currentColor" aria-hidden="true" display="inline-block" />
        </CloseButton>
      )}
    </Box>
  );
};

Pill.displayName = 'Pill';

export default Pill;
