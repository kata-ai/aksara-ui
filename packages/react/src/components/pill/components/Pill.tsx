import * as React from 'react';
import { IconClose } from '@aksara-ui/icons';

import { Box } from '../../../layout';
import { useComponentStyles } from '../../../system';
import { UnstyledButton, UnstyledButtonProps } from '../../button';

export interface PillProps extends Omit<UnstyledButtonProps, 'size'>, React.ComponentPropsWithoutRef<'button'> {
  /** The variant options available for a pill. */
  variant?: 'default' | 'info' | 'critical' | 'success' | 'active';
  /** Additional CSS classes to give to the component. */
  className?: string;
  /** Additional CSS properties to give to the component. */
  style?: React.CSSProperties;
  /** Add a custom icon to the left of the icon. */
  icon?: React.ReactNode;
  /** Set `true` to display the Close icon in the Pill. */
  withCloseIcon?: boolean;
  /** Triggers when the pill is clicked. */
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
}

const Pill = React.forwardRef<HTMLButtonElement, PillProps>(
  (
    {
      className,
      style,
      children,
      variant = 'default',
      icon,
      withCloseIcon = false,
      onClick,
      sx,
      type = 'button',
      ...rest
    },
    ref
  ) => {
    const boxStyles = useComponentStyles('pillRoot', { variant });

    return (
      <UnstyledButton
        ref={ref}
        style={style}
        className={className}
        sx={{ ...boxStyles, ...sx }}
        type={type}
        onClick={onClick}
        {...rest}
      >
        {icon}
        <Box as="span" display="inline-blick" px="xxs" fontSize="12px" lineHeight="18px">
          {children}
        </Box>
        {withCloseIcon && (
          <Box display="flex" alignItems="center" justifyContent="center">
            <IconClose size={12} fill="currentColor" aria-hidden="true" display="inline-block" />
          </Box>
        )}
      </UnstyledButton>
    );
  }
);

Pill.displayName = 'Pill';

export default Pill;
