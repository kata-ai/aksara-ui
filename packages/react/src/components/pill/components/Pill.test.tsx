import * as React from 'react';
import { render } from '@testing-library/react';
import Pill from './Pill';

describe('components/Pill', () => {
  describe('Pill', () => {
    test('renders correctly', () => {
      const { container } = render(<Pill>test pill</Pill>);

      expect(container).toHaveTextContent('test pill');
    });
  });
});
