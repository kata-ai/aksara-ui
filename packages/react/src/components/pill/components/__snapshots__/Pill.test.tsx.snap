// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Pill Pill renders with additional variants 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border-radius: 9999px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  padding-left: xs;
  padding-right: xs;
  padding-top: 8px;
  padding-bottom: 8px;
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    test pill
  </span>
</div>
`;
